src/Dfc.CourseDirectory.Web/Controllers/BulkUploadApprenticeshipsController.cs

```diff
@@ -297,16 +297,19 @@ public async Task<IActionResult> DeleteFile(DeleteFileViewModel model)
                 var archiveFilesResult = _blobService.ArchiveFiles($"{UKPRN.ToString()}{BlobContainerPath}");
             }
 
-            var deleteBulkuploadResults = await _apprenticeshipService.DeleteBulkUploadApprenticeships(UKPRN);
-
-            if (deleteBulkuploadResults.IsSuccess)
+            var apprenticeships = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetApprenticeships
             {
-                return RedirectToAction("DeleteFileConfirmation", "BulkUploadApprenticeships", new { fileUploadDate = fileUploadDate });
-            }
-            else
+                Predicate = a =>
+                    a.ProviderUKPRN == UKPRN
+                    && (a.RecordStatus == (int)ApprenticeshipStatus.BulkUploadPending || a.RecordStatus == (int)ApprenticeshipStatus.BulkUploadReadyToGoLive)
+            });
+
+            foreach (var apprenticeship in apprenticeships.Values)
             {
-                return RedirectToAction("Index", "Home", new { errmsg = "Delete All Bulk Uploaded Apprenticeships Error" });
+                await _cosmosDbQueryDispatcher.ExecuteQuery(new DeleteApprenticeship { ApprenticeshipId = apprenticeship.Id, ProviderUkprn = apprenticeship.ProviderUKPRN });
             }
```

Comment:
Does this need to include the functionality below too?
https://github.com/SkillsFundingAgency/dfc-providerportal-apprenticeships/blob/main/src/Dfc.ProviderPortal.Apprenticeships/Functions/DeleteBulkUploadApprenticeships.cs#L40-L44

Reply:
I did contemplate that, but I tried to rationalise why that was in there in the first place.

As far as I'm aware, we don't ever set the `Apprenticeship.RecordStatus` to `MigrationPending` or `MigrationReadyToGoLive`, do we?

This seems to me like a hack to clean up some apprenticeships with legacy statuses.

Reply:
We don't set any new records to that status but we may still have some old ones around.

Reply:
Should cancelling an apprenticeships bulk upload really trigger that process? If the intention is to only clean up old records and nothing to do with the bulk upload process, then I vote to omit these behaviours.

Reply:
It doesn't make sense to me (I can understand it when publishing a bulk upload but not when deleting pending records). If you're happy to take the behaviour change then I'm fine with it.